src/isar_exr/robotinterface.py

```diff
@@ -231,9 +242,53 @@ def initiate_step(self, step: Step) -> None:
         raise NotImplementedError
 
     def step_status(self) -> StepStatus:
-        # TODO: use currentMissionExecution(robotID)
-        self.logger.error("An invalid interface function was called")
-        raise NotImplementedError
+        try:
+            (mission_status, current_task_id) = (
+                self.api.get_mission_status_and_current_task(settings.ROBOT_EXR_ID)
+            )
+            step_status: StepStatus = ExrStepStatus(mission_status).to_step_status()
+        except NoMissionRunningException:
+            # This is a temporary solution until we have mission status by mission id
+            return MissionStatus.Successful
+        except Exception as e:
+            message: str = "Could not get status of running mission\n"
+            self.logger.error(message)
+            raise RobotMissionStatusException(
+                error_description=message,
+            )
+
+        if current_task_id == "end_mission":
+            return StepStatus.Successful
+
+        if (
+            current_task_id == "start_mission"
+            or current_task_id == None
+            or current_task_id == ""
+        ):
+            return StepStatus.NotStarted
+
+        try:
+            step: List[str] = list(
+                filter(lambda step: current_task_id in step, self.mission_task_ids)
+            )[0]
+            task_index: int = self.mission_task_ids.index(step)
```

Comment:
Can't this just be simplified to the suggestion below?
```suggestion
            task_index: int = self.mission_task_ids.index(current_task_id)
```

Reply:
In this case self.mission_task_ids is a 2D array, where the first dimension is the task and the inner dimension is the steps within that task. This is maybe not a good name for the object; it has the name since it started off as an array of task IDs. Since we are trying to find the index which contains a task which in turn contains current_task_id as a step, we need to first do this filtering to find the right task based on its steps, before we can use it to find the right task index.